### PR TITLE
Describe params

### PR DESCRIPTION
**Pull request by navFooh**
- Adds fallback for decoding v1-7 actor codes
- Converts actor method numbers in JSON from string to number
- Updated actor descriptors to treat uint8 array as base64 strings
- Makes supplying network name to utility functions mandatory
- `getActorName` now accepts the actor code as `string` or `LotusCID`
- Improves verbosity of errors while applying descriptor files
- Add utility to retrieve the method name from the actor descriptors
- Add utility for describing message params
- Updated tests

### CODE REVIEW

File: packages/filecoin-actor-utils/src/utils/code.ts [hunk 1]
```diff
@@ -1,44 +1,53 @@
...
-  networkName?: NetworkName
+  actorCode: ActorCode | LotusCID,
+  networkName: NetworkName
 ): ActorName | null => {
-  // If "networkName" is provided, return the actor name
-  // for that specific network or null when not found
-  if (networkName) {
-    return networkActorCodeMapInv[networkName]?.[actorCode] ?? null
-  }
+  const code = typeof actorCode === 'string' ? actorCode : actorCode['/']
```
Comment (Schwartz10) [inline packages/filecoin-actor-utils/src/utils/code.ts]: Typescript would probably catch this, but if we ever didn't pass a CID, or a CID was `null` that accidentally got passed through to this function, then `actorCode['/']` would result in an unhandled runtime error. Probably safe to just `actorCode?.['/']`

File: packages/filecoin-actor-utils/test/index.spec.ts [hunk 2]
```diff
@@ -34,7 +33,7 @@ describe('utils', () => {
 
     test('should find the mainnet actor code when not providing a network', () => {
```
Comment (Schwartz10) [inline packages/filecoin-actor-utils/test/index.spec.ts]: Might be better to change the name of this tests, as now you are finding a code

**Review by Schwartz10 — CHANGES_REQUESTED** [review]
One thought moving forward - only `actorNames` apply to built-in actors. FEVM actors, and native (user programmed) FVM actors are not going to have/need the concept of an "actorName", besides, potentially certain things that fit a specific agreed upon standard.

I.e. - an ERC721 or an ERC20 actor might be named that way. Not sure if this is an important observation, but just thought I'd mention it here. Just one place where i think we can protect against a potential unhandled runtime error

**Comment by navFooh** [comment]
> One thought moving forward - only `actorNames` apply to built-in actors. FEVM actors, and native (user programmed) FVM actors are not going to have/need the concept of an "actorName", besides, potentially certain things that fit a specific agreed upon standard.
> 
> I.e. - an ERC721 or an ERC20 actor might be named that way. Not sure if this is an important observation, but just thought I'd mention it here. Just one place where i think we can protect against a potential unhandled runtime error

That's a good point and important to keep in mind. This also seems to tie into the question of how we're going to describe F(E)VM actors in the first place. But it seems to makes sense to tie a descriptor directly to an actor code instead of the name, since that would also make sure that the descriptors are specific for that actor version.

Let's revise when we circle back to this scope of work.

File: packages/filecoin-actor-utils/src/utils/code.ts (hunk 1 above)
Comment (navFooh) [reply packages/filecoin-actor-utils/src/utils/code.ts]: Good call

File: packages/filecoin-actor-utils/test/index.spec.ts (hunk 2 above)
Comment (navFooh) [reply packages/filecoin-actor-utils/test/index.spec.ts]: Also good call

**Pull request merged**